common/src/main/java/net/william278/huskclaims/claim/ClaimWorld.java

```diff
@@ -460,6 +460,33 @@ public boolean isBannedFromClaim(@NotNull OnlineUser user, @NotNull Claim claim,
         return false;
     }
 
+    // Check if a claim is private for user
+    @ApiStatus.Internal
+    public boolean canAccessPrivateClaim(@NotNull OnlineUser user, @NotNull Claim claim, @NotNull HuskClaims plugin) {
+        if (!plugin.getSettings().getClaims().getBans().isPrivateClaims()) {
+            return true;
+        }
+
+        if (!claim.isPrivateClaim()) {
+            return false;
+        }
+        
+        Optional<UUID> owner = claim.getOwner();
+        if (owner.isPresent() && owner.get().equals(user.getUuid())) {
+            return false;
+        }
+
+        if (isIgnoringClaims(user, plugin)) {
+            return false;
+        }
+        
+        if (claim.getTrustLevel(user, plugin).isPresent()) {
+            return false;
+        }
+
+        return true;
```

Comment:
Just a bit of a cleanup here to simplify things.
```suggestion
        final Optional<UUID> owner = claim.getOwner();
        if (owner.isPresent() && owner.get().equals(user.getUuid()) || isIgnoringClaims(user, plugin)) {
            return false;
        }
        return claim.getTrustLevel(user, plugin).isEmpty();
```
I've also realised this method name is now inconsistent with the boolean value we're returning - so perhaps lets go with `isBlockedFromPrivateClaim`

Reply:
Changed it :)